libs/qlik/src/dashboard/vl-qlik-dashboard.component.ts

```diff
@@ -0,0 +1,315 @@
+import { CSSResult, html, PropertyDeclarations } from 'lit';
+import { BaseLitElement, registerWebComponents, webComponent } from '@domg-wc/common-utilities';
+import { stardust } from '@nebula.js/stardust/types';
+import { Qlik, STARDUST } from '@domg/qlik-lib';
+import { bindVlSelect, debounce, queryById, queryBySelectorAll, renderStack } from '../utils/qlik-render-utils';
+import { DashboardVisualization, Filter } from '../utils/vl-qlik.model';
+import { vlElementsStyle } from '@domg-wc/elements';
+import dashboardUigStyle from './vl-qlik-dashboard.uig-css';
+import { VlQlikVisualComponent } from '../visual';
+import { VlLoaderComponent, VlPillComponent } from '@domg-wc/components';
+
+@webComponent('vl-qlik-dashboard')
+export class VlQlikDashboardComponent extends BaseLitElement {
+    visuals: DashboardVisualization[][];
+    filters: Filter[] = [];
+    connection?: Qlik;
+    private connected = false;
+    private initialized = false;
+    private filtersLoading = true;
+    private stardust?: stardust.Embed;
+    private selected = {};
+
+    static {
+        registerWebComponents([VlQlikVisualComponent, VlLoaderComponent, VlPillComponent]);
+    }
+
+    static get styles(): (CSSResult | CSSResult[])[] {
+        return [vlElementsStyle, dashboardUigStyle];
+    }
+
+    static get properties(): PropertyDeclarations {
+        return {
+            visuals: { type: Array },
+            filters: { type: Array },
+            connection: { type: Object },
+            connected: { type: Boolean, state: true },
+            initialized: { type: Boolean, state: true },
+            filtersLoading: { type: Boolean, state: true },
+            selected: { type: Object, state: true },
+        };
+    }
+
+    async firstUpdated(_changedProperties) {
+        super.firstUpdated(_changedProperties);
+        if (this.connection && !this.connected) {
+            this.stardust = await STARDUST(this.connection.app);
+            this.connected = true;
+            this.dispatchEvent(new CustomEvent('connected'));
+        }
+        if (this.connected && this.filters && this.filters.length > 0) {
+            try {
+                await Promise.all(this.filters.map((f) => this.connection.addFilters(f.filter)));
+            } catch (e) {
+                // do nothing
+            }
+        }
+        this.initialized = true;
+        this.dispatchEvent(new CustomEvent('initialized'));
+        this.search();
+    }
+
+    async updated(_changedProperties) {
+        super.updated(_changedProperties);
+        if (this.initialized && this.filters && this.filters.length > 0) {
+            this.bindFilters();
```

Comment:
`async` voor `updated` niet nodig gezien je geen `await` zet voor `this.bindFilters();`
- of `await this.bindFilters();` zetten
- of `async`-keyword verwijderen